test/common/pixel-tests

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -eux
+
+SUBDIR=test/reference
+
+GITHUB_BASE="${GITHUB_BASE:-cockpit-project/cockpit}"
+GITHUB_REPOSITORY="${GITHUB_BASE%/*}/pixel-test-reference"
+CLONE_REMOTE="https://github.com/${GITHUB_REPOSITORY}"
+PUSH_REMOTE="git@github.com:${GITHUB_REPOSITORY}"
+
+message() {
+    [ "${V-}" != 0 ] || printf "  %-8s %s\n" "$1" "$2"
+}
+
+get_index_gitlink() {
+    if ! git ls-files -s "$1" | egrep -o '\<[[:xdigit:]]{40}\>'; then
+        echo "*** couldn't read gitlink for file $1 from the index" >&2
+        exit 1
+    fi
+}
+
+subgit() {
+    git -C "$SUBDIR" "$@"
+}
+
+setup() {
+    # Cases:
+    #
+    # - $SUBDIR doesn't exist in index => fail
+    # - $SUBDIR is no gitlink in index => fail
+    # - $SUBDIR doesn't exist in fsys or is dir but $SUBDIR/.git doesn't exist => init repo
+    # - $SUBDIR is not git repo => fail
+
+    [ -d "$SUBDIR" ] || mkdir -p "$SUBDIR"
+    [ -d "$SUBDIR/.git" ] || subgit init
+}
+
+cmd_pull() {
+    setup || exit 1
+
+    local sha="$(get_index_gitlink $SUBDIR)"
+    local tag="sha-${sha}"
+
+    echo PULL $tag
+
+    subgit fetch "$CLONE_REMOTE" "$tag"
+    subgit checkout "$sha"
+}
+
+cmd_status() {
+    setup || exit 1
+
+    subgit status
+}
+
+cmd_push() {
+    setup || exit 1
+
+    subgit checkout --orphan temp
+    subgit commit -a -m "$(date)"
```

Comment:
What does that actually commit? Nothing gets added or changed anywhere? 

Also, why the dance with the orphan checkout and such? It could just always init an empty repo, commit, and just push the tag, like in the [c-machines publish-dist repo](https://github.com/cockpit-project/cockpit-machines/blob/main/.github/workflows/publish-dist.yml#L33)

Reply:
Yes, this was way too naive.

Reply:
Ok, I have made this even more complex now. The difference between this and publish-dist and node_modules is that test/references contains files that are directly worked on by the developer. So I didn't want to assume that we can just blow away the repository and start from scratch.

Also, I want `push` to be a no-op if nothing has actually changes, so we need to keep some history around. So I am not sure how much simpler this will get.

The one wart is the necessity to create the "temp" branch and the circus necessary to get rid of it again.  The pixel-test-reference repository has a branch called "empty", and we could use that as the parent of all commits... that would probably lead to simpler code.